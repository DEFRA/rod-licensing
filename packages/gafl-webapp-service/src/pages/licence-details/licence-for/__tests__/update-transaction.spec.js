import updateTransaction from '../update-transaction'

describe('licence-for > update-transaction', () => {
  beforeEach(jest.clearAllMocks)

  const mockPageCacheGet = jest.fn()
  const mockStatusCacheSet = jest.fn()

  const mockRequest = {
    cache: () => ({
      helpers: {
        page: {
          getCurrentPermission: mockPageCacheGet
        },
        status: {
          setCurrentPermission: mockStatusCacheSet
        }
      }
    })
  }

  describe('default', () => {
    it('should set isLicenceForYou to true if user has selected you', async () => {
      mockPageCacheGet.mockImplementationOnce(() => ({payload:{'licence-for': 'you'}}))
      await updateTransaction(mockRequest)
      expect(mockStatusCacheSet).toHaveBeenCalledWith({ isLicenceForYou: true })
    })

    it('should set isLicenceForYou to false if user has selected someone-else', async () => {
      mockPageCacheGet.mockImplementationOnce(() => ({payload:{'licence-for': 'someone-else'}}))
      await updateTransaction(mockRequest)
      expect(mockStatusCacheSet).toHaveBeenCalledWith({ isLicenceForYou: false })
    })
  })
})
