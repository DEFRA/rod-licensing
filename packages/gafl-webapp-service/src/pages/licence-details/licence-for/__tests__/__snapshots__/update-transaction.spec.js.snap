// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`licence-for > update-transaction default should clear personal information from _current_ permission if licence for has changed from someone-else to you 1`] = `
Object {
  "licence-for": Object {
    "payload": Object {
      "continue": "",
      "licence-for": "you",
    },
  },
}
`;

exports[`licence-for > update-transaction default should clear personal information from _current_ permission if licence for has changed from you to someone-else 1`] = `
Object {
  "licence-for": Object {
    "payload": Object {
      "continue": "",
      "licence-for": "someone-else",
    },
  },
}
`;

exports[`licence-for > update-transaction default should clear personal information from session if licence for has changed from someone-else to you 1`] = `
Object {
  "licence-for": Object {
    "payload": Object {
      "continue": "",
      "licence-for": "you",
    },
  },
}
`;

exports[`licence-for > update-transaction default should clear personal information from session if licence for has changed from you to someone-else 1`] = `
Object {
  "licence-for": Object {
    "payload": Object {
      "continue": "",
      "licence-for": "someone-else",
    },
  },
}
`;
